README.md

```diff
@@ -243,6 +242,56 @@ In the strict mode, `async` functions are checked such that they only raise
 effects on forward declarations, callbacks and methods using
 `{.raises: [CatchableError].}` (or more strict) annotations.
 
+### Cancellation support
+
+Any running `Future` can be cancelled.
+
+```nim
+proc c1 {.async.} =
+  echo "Before sleep"
+  try:
+    await sleepAsync(10.minutes)
+    echo "After sleep" # never reached
+  except CancelledError as exc:
+    echo "We got cancelled!"
+    raise exc
+
+proc c2 {.async.} =
+  await c1()
+  echo "Never reached, since the CancelledError bubble up"
+
+let work = c2()
+waitFor(work.cancelAndWait())
+```
+
+The cancellation will travel down the call stack, and throw a `CancelledError`
+on the currently running `await` (in this case the `await sleepAsync` in `c1`)
+
+The `CancelledError` will now travel up the stack like any other exception.
+It can be caught and handled (for instance, freeing some resources), or it can
+swallowed if a task must not be cancelled.
+
+Since `CancelledError` inherits from `CatchableError`, be careful to handle it
+appropriately:
+
+```nim
+proc c3 {.async.} =
+  try:
+    await somethingThatCanFail()
+  except CatchableError as exc:
+    echo "Something went wrong: ", exc.msg
+
+proc c4 {.async.} =
+  while true:
+    await c3()
+
+waitFor(c4().wait(10.seconds))
+```
+
+In this example, `c3` swallows cancellations, so `c4` cannot be cancelled
+properly, and the `wait` timeout will not work
+
+
 ### Multiple async backend support
```

Comment:
This paragraph should be removed.
Because there is no multiple backends, we trying to make the only one backend and if some other packages support `asyncdispatch` it their own way. Chronos is not going to support or be compatible with `asyncdispatch`. Also i do not see any reasons to have links in here.

Reply:
cc @arnetheduck who added it here

Reply:
chronos does not maintain compatibility with ad but there exist libraries that support both ad and chronos by providing an abstraction layer that removes the difference - they need a way to pick which backend to use, and it's better that there exists a standard way of doing so than not - this is important also when frameworks want to introduce support for chronos without breaking their existing ad support, for example to provide a transition period - as such, this is documented both here and in the standard library